addons/hr_holidays/models/hr_leave.py

```diff
@@ -1668,43 +1672,29 @@ def _get_attendances(self, request_date_from, request_date_to, day_period=None):
         if day_period:
             domain.append(('day_period', '=', day_period))
         attendances = self.env['resource.calendar.attendance']._read_group(domain,
-            ['week_type', 'dayofweek', 'day_period'],
+            ['week_type', 'dayofweek'],
             ['hour_from:min', 'hour_to:max'])
 
         # Must be sorted by dayofweek ASC and day_period DESC
-        attendances = sorted([DummyAttendance(hour_from, hour_to, dayofweek, day_period, week_type) for week_type, dayofweek, day_period, hour_from, hour_to in attendances], key=lambda att: (att.dayofweek, att.day_period != 'morning'))
+        attendances = sorted([DummyAttendance(hour_from, hour_to, dayofweek, week_type) for week_type, dayofweek, hour_from, hour_to in attendances], key=lambda att: att.dayofweek)
 
-        default_value = DummyAttendance(0, 0, 0, 'morning', False)
+        # If we can't find any attendances on the exact days of the request,
+        # we default to the widest possible range that exists in the schedule.
+        default_start = min(map(lambda att: att.hour_from, attendances), default=0)
+        default_end = max(map(lambda att: att.hour_to, attendances), default=0)
```

Comment:
What's the added value of doing that instead of taking 0? If there are attendances on the day of the start/end, it will be used for hour_from/hour_to, and if there are none the time won't matter anyway :thinking:

Reply:
I think the spirit of this method was to take starting and ending hours that look natural, given the schedule the employee has. For example, if I work 9-5 every day except for tuesday, but take my time off from tuesday to thursday, it indeed doesn't matter what time the time off actually starts, but it will look nicer on the calendar if it starts at 9 as usual, instead of it starting at midnight all of the sudden. 

But yeah, maybe this doesn't really matter, and maybe we should just prevent people from taking time off that starts or ends on days they are not really working. But this is a more of a fix that fixes current problematic behaviour without really changing the intended functionality.